# Assignment 05
https://github.com/wcj365/python-stats-dataviz/blob/master/assignment_fall2020/assignment_05.md

## Step 1
Explore College Scorecard website - https://collegescorecard.ed.gov/

## Step 2
Explore College Scorecard Data web page - https://collegescorecard.ed.gov/data/

## Step 3
Download and read the documentation - https://collegescorecard.ed/gov/assets/FullDataDocumentation.pdf

## Step 4
Download the dataset - https://ed-public-download.app.cloud.gov/downloads/CollegeScorecard_Raw_Data.zip (265MB Zip File)

## Step 5
Unzip the file into a folder in your personal computer (there are many files, one for each school year)

## Step 6
Upload the 2017-2018 school year file "MERGED2017_18_PP.csv" to Colab and create a Jupyter notebook to explore the dataset:
- how many colleges or rows?
- how many variables or columns?
- look at the first 5 colleges
- look at the last 5 colleges
- look at a random sample of 5 colleges

In [1]:
import pandas as pd

# read in csv as a dataframe
df = pd.read_csv('MERGED2017_18_PP.csv')

In [2]:
# view dataframe
df.head(10)

UNITID    OPEID  ...  FTFTPCTFLOAN_POOLED_SUPP SCUGFFN_POOLED
0  100654   100200  ...                       NaN            NaN
1  100663   105200  ...                       NaN            NaN
2  100690  2503400  ...                       NaN            NaN
3  100706   105500  ...                       NaN            NaN
4  100724   100500  ...                       NaN            NaN
5  100751   105100  ...                       NaN            NaN
6  100760   100700  ...                       NaN            NaN
7  100812   100800  ...                       NaN            NaN
8  100830   831000  ...                       NaN            NaN
9  100858   100900  ...                       NaN            NaN

[10 rows x 1986 columns]

In [4]:
# How many rows and columns are in the dataframe?
df.shape

(6643, 1986)

There are 6643 rows and 1986 columns in the dataframe.

There are 6643 colleges and 1986 variables.

In [5]:
# Look at the first 5 colleges
df.head(5)

UNITID    OPEID  ...  FTFTPCTFLOAN_POOLED_SUPP SCUGFFN_POOLED
0  100654   100200  ...                       NaN            NaN
1  100663   105200  ...                       NaN            NaN
2  100690  2503400  ...                       NaN            NaN
3  100706   105500  ...                       NaN            NaN
4  100724   100500  ...                       NaN            NaN

[5 rows x 1986 columns]

In [7]:
# Look at the last 5 colleges
df.tail(5)

UNITID    OPEID  ...  FTFTPCTFLOAN_POOLED_SUPP SCUGFFN_POOLED
6638  490674  4125502  ...                       NaN            NaN
6639  490692  4254900  ...                       NaN            NaN
6640  490717  4269800  ...                       NaN            NaN
6641  490735  4168701  ...                       NaN            NaN
6642  490771   458620  ...                       NaN            NaN

[5 rows x 1986 columns]

In [8]:
# Look at a random sample of 5 colleges
df.sample(5)

UNITID    OPEID  ...  FTFTPCTFLOAN_POOLED_SUPP SCUGFFN_POOLED
4973  442727   982802  ...                       NaN            NaN
4767  434830  3319300  ...                       NaN            NaN
1380  159522  2117800  ...                       NaN            NaN
366   117946  1164900  ...                       NaN            NaN
1387  159939   201500  ...                       NaN            NaN

[5 rows x 1986 columns]

## Step 7
Since there are too many variables to explore, many of which are not interesting. Create a smaller dataframe with the following 9 variables:
- UNITID
- INSTNM
- CITY
- STABBR
- ZIP
- ADM_RATE
- UGDS
- TUITIONFEE_IN
- MN_EARN_WNE_P10

In [9]:
df_smaller = df[['UNITID','INSTNM','CITY','STABBR','ZIP','ADM_RATE','UGDS','TUITIONFEE_IN','MN_EARN_WNE_P10']]

In [10]:
# View smaller dataframe
df_smaller.head()

UNITID                               INSTNM  ... TUITIONFEE_IN MN_EARN_WNE_P10
0  100654             Alabama A & M University  ...        9857.0             NaN
1  100663  University of Alabama at Birmingham  ...        8328.0             NaN
2  100690                   Amridge University  ...        6900.0             NaN
3  100706  University of Alabama in Huntsville  ...       10280.0             NaN
4  100724             Alabama State University  ...       11068.0             NaN

[5 rows x 9 columns]

## Step 8
Download and read the data dictionary file - https://collegescorecard.ed.gov/assets/CollegeScorecardDataDictionary.xlsx

In [11]:
df_dict = pd.read_csv('dictionary.csv')
df_dict.head()

NAME OF DATA ELEMENT dev-category  ... Unnamed: 9 Unnamed: 10
0         Unit ID for institution         root  ...        NaN         NaN
1  8-digit OPE ID for institution         root  ...        NaN         NaN
2  6-digit OPE ID for institution         root  ...        NaN         NaN
3                Institution name       school  ...        NaN         NaN
4                            City       school  ...        NaN         NaN

[5 rows x 11 columns]

## Step 9
Find out the definition of the above 9 variables and document the definitions in the Jupyter Notebook in a Markdown cell so that the readers of your notebook will know the meaning of these variables.

| Variable Name | Notes |
| --- | --- |
| UNITID | Unit ID for institution |
| INSTNM | Institution name |
| CITY | City |
| STABBR | State postcode |
| ZIP | ZIP code |
| ADM_RATE | Admission rate |
| UGDS | Enrollment of undergraduate certificate/degree-seeking students |
| TUITIONFEE_IN | In-state tuition and fees |
| MN_EARN_WNE_P10 | Mean earning of students working and not enrolled 10 years after entry |




## Step 10
Explore the smaller data frame:
- look at the first 10 colleges
- look at the last 10 colleges
- look at a random sample of 10 colleges
- Find and display UMBC
- Find Univ. of Maryland College Park
- Find and display Loyola College Maryland
- Find and display Johns Hopkins
- Calculate the min, max, and average in-state tuition of the above four instiutions

In [12]:
# look at the first 10 colleges
df_smaller.head(10)

UNITID                               INSTNM  ... TUITIONFEE_IN MN_EARN_WNE_P10
0  100654             Alabama A & M University  ...        9857.0             NaN
1  100663  University of Alabama at Birmingham  ...        8328.0             NaN
2  100690                   Amridge University  ...        6900.0             NaN
3  100706  University of Alabama in Huntsville  ...       10280.0             NaN
4  100724             Alabama State University  ...       11068.0             NaN
5  100751            The University of Alabama  ...       10780.0             NaN
6  100760    Central Alabama Community College  ...        4440.0             NaN
7  100812              Athens State University  ...           NaN             NaN
8  100830      Auburn University at Montgomery  ...        8020.0             NaN
9  100858                    Auburn University  ...       10968.0             NaN

[10 rows x 9 columns]

In [13]:
# look at the last 10 colleges
df_smaller.tail(10)

UNITID  ... MN_EARN_WNE_P10
6633  490522  ...             NaN
6634  490531  ...             NaN
6635  490559  ...             NaN
6636  490568  ...             NaN
6637  490601  ...             NaN
6638  490674  ...             NaN
6639  490692  ...             NaN
6640  490717  ...             NaN
6641  490735  ...             NaN
6642  490771  ...             NaN

[10 rows x 9 columns]

In [14]:
# look at a random sample of 10 colleges
df_smaller.sample(10)

UNITID  ... MN_EARN_WNE_P10
2476  197984  ...             NaN
2172  188146  ...             NaN
1899  177816  ...             NaN
6291  484057  ...             NaN
3913  241216  ...             NaN
5033  444088  ...             NaN
2866  208017  ...             NaN
236   110705  ...             NaN
5998  476063  ...             NaN
2047  183035  ...             NaN

[10 rows x 9 columns]

In [24]:
# Find and display UMBC
df_bc = df_smaller.loc[df_smaller['INSTNM'] == 'University of Maryland-Baltimore County']
df_bc

UNITID  ... MN_EARN_WNE_P10
1478  163268  ...             NaN

[1 rows x 9 columns]

In [25]:
# Find and display Univ. of Maryland College Park
df_cp = df_smaller.loc[df_smaller['INSTNM'] == 'University of Maryland-College Park']
df_cp

UNITID  ... MN_EARN_WNE_P10
1479  163286  ...             NaN

[1 rows x 9 columns]

In [26]:
# Find and display Loyola College Maryland
df_l = df_smaller.loc[df_smaller['INSTNM'] == 'Loyola University Maryland']
df_l

UNITID                      INSTNM  ... TUITIONFEE_IN MN_EARN_WNE_P10
1475  163046  Loyola University Maryland  ...       47560.0             NaN

[1 rows x 9 columns]

In [27]:
# Find and display Johns Hopkins
df_jh = df_smaller.loc[df_smaller['INSTNM'] == 'Johns Hopkins University']
df_jh

UNITID                    INSTNM  ... TUITIONFEE_IN MN_EARN_WNE_P10
1473  162928  Johns Hopkins University  ...       52170.0             NaN

[1 rows x 9 columns]

In [28]:
# Create a dataframe with the above four instiutions
df_MD = pd.concat([df_bc,df_cp,df_l,df_jh], axis=0)
df_MD.head()

UNITID  ... MN_EARN_WNE_P10
1478  163268  ...             NaN
1479  163286  ...             NaN
1475  163046  ...             NaN
1473  162928  ...             NaN

[4 rows x 9 columns]

In [29]:
# Calculate the min
print("Min of TUITIONFEE_IN:", df_MD['TUITIONFEE_IN'].min())

# Calculate the max
print("Max of TUITIONFEE_IN:", df_MD['TUITIONFEE_IN'].max())

# Calculate the average
print("Average of TUITIONFEE_IN:", df_MD['TUITIONFEE_IN'].mean())

Min of TUITIONFEE_IN: 10399.0
Max of TUITIONFEE_IN: 52170.0
Average of TUITIONFEE_IN: 30411.75


## Step 11
Filter the smaller dataset further to only keep colleges from Maryland, Virgina, Washington DC, Delaware, and Pennsylvinia. Save the filtered dataset to a new file named "Five_States_Colleges.csv" for future analysis. Upload this file to your GitHub along with the notebook.

In [33]:
# filter based on 5 states
df_five = df_smaller.loc[(df_smaller['STABBR'] == 'MD') | (df_smaller['STABBR'] == 'VA') | (df_smaller['STABBR'] == 'DC') | (df_smaller['STABBR'] == 'DE') | (df_smaller['STABBR'] == 'PA')]
df_five.head()

UNITID  ... MN_EARN_WNE_P10
634  130837  ...             NaN
635  130873  ...             NaN
636  130907  ...             NaN
637  130934  ...             NaN
638  130943  ...             NaN

[5 rows x 9 columns]

In [34]:
# export dataframe to csv
df_five.to_csv('Five_States_Colleges.csv')